packages/app-desktop/ElectronAppWrapper.ts

```diff
@@ -352,6 +362,16 @@ export default class ElectronAppWrapper {
 		this.electronApp_.on('activate', () => {
 			this.win_.show();
 		});
+
+		this.electronApp_.on('open-url', (_event: any, url: string) => {
+			void this.openUrl(url);
+		});
+	}
+
+	async openUrl(url: string) {
+		this.win_.webContents.send('asynchronous-message', 'openUrl', {
+			url: url,
+		});
```

Comment:
I wonder if the all the message passing is necessary? This class could simply save the intial URL to a class property, then it can be accessed from the main screen when it's ready using `bridge().electronApp()`

Reply:
I didn't know I could do this. This helps in the case when Joplin is not running and is opened from a url. However when Joplin is running this isn't going to work.